File: sharedpreference/src/main/java/com/seatrend/sharedpreference/MainActivity.java
package com.seatrend.sharedpreference;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

    EditText edit1;
    EditText edit2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        edit1 = findViewById(R.id.edit1);
        edit2 = findViewById(R.id.edit2);
    }

    public void save(View view) {
        SharedPerfrenceUtils.save("edit1", edit1.getText().toString());
    }

    public void read(View view) {
        edit2.setText(SharedPerfrenceUtils.get("edit1"));
    }

    public void deleteALL(View view) {
        SharedPerfrenceUtils.deleteAll();
    }
}

File: LockScreen_AD/src/main/java/com/laiyu/android_demo1/MainActivity.kt
package com.laiyu.android_demo1

import android.annotation.SuppressLint
import android.content.Context
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.os.PowerManager
import android.provider.Settings
import android.provider.Settings.ACTION_MANAGE_OVERLAY_PERMISSION
import android.view.View
import android.view.Window
import android.view.WindowManager
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {

        // 设置点亮屏幕


        super.onCreate(savedInstanceState)

        val win: Window = window
        win.addFlags(
            WindowManager.LayoutParams.FLAG_SHOW_WHEN_LOCKED
                    or WindowManager.LayoutParams.FLAG_DISMISS_KEYGUARD
                    or WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON
                    or WindowManager.LayoutParams.FLAG_TURN_SCREEN_ON
        )

        Uitls.showLog("MainActivity 被创建")

        setContentView(R.layout.activity_main)
        canDisplay()

    }

    override fun onDestroy() {
        super.onDestroy()
        Uitls.showLog("MainActivity 被销毁")
    }

    @SuppressLint("InvalidWakeLockTag")
    override fun onNewIntent(intent: Intent?) {
        super.onNewIntent(intent)
        val pm =
            this.getSystemService(Context.POWER_SERVICE) as PowerManager
        if (!pm.isScreenOn) {
            val wl = pm.newWakeLock(
                (PowerManager.ACQUIRE_CAUSES_WAKEUP or
                        PowerManager.SCREEN_BRIGHT_WAKE_LOCK), "bright"
            )
            wl.acquire()
            wl.release()
        }
    }

    fun btn(view: View) {
        MyApp.mContext!!.startService(Intent(MyApp.mContext, ADService::class.java))
    }

    private fun canDisplay() {
        if (!Settings.canDrawOverlays(this)) {
            Toast.makeText(this, "当前无权限，请授权", Toast.LENGTH_SHORT)
            startActivityForResult(
                Intent(
                    ACTION_MANAGE_OVERLAY_PERMISSION,
                    Uri.parse("package:$packageName")
                ), 0
            )
        }
    }
}